stl/inc/ranges

```diff
@@ -29,6 +29,69 @@ namespace ranges {
     template <class _Rng>
     concept viewable_range = range<_Rng>
         && (borrowed_range<_Rng> || view<remove_cvref_t<_Rng>>);
+
+
+    // CLASS ranges::ref_view
+    // clang-format off
+    template <range _Rng>
+        requires is_object_v<_Rng>
+    class ref_view : public view_interface<ref_view<_Rng>> {
+        // clang-format on
+        void _Rvalue_poison(_Rng&);
+        void _Rvalue_poison(_Rng&&) = delete;
+
+    public:
+        constexpr ref_view() noexcept = default;
+
+        // clang-format off
+        template <_Not_same_as<ref_view> _OtherRng>
+        //     requires convertible_to<_OtherRng, _Rng&> && requires { _Rvalue_poison(declval<_OtherRng>()); }
+        constexpr ref_view(_OtherRng&& _Range)
+            // clang-format on
+            : _Myrange(_STD addressof(static_cast<_Rng&>(_STD forward<_OtherRng>(_Range)))) {}
+
+        constexpr _Rng& base() const {

```

Comment:
I believe these member functions should be audited for `_NODISCARD` (it looks like "all of them" but I am not 100% sure).

Reply:
Went for all of them as they all are const accessors